android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt

```diff
@@ -0,0 +1,187 @@
+package com.solana.mobilewalletadapter.clientlib
+
+import android.net.Uri
+import android.util.Log
+import androidx.activity.result.ActivityResultCaller
+import com.solana.mobilewalletadapter.clientlib.protocol.JsonRpc20Client
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+import com.solana.mobilewalletadapter.clientlib.scenario.LocalAssociationScenario
+import com.solana.mobilewalletadapter.clientlib.scenario.Scenario
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioChooser
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioTypes
+import com.solana.mobilewalletadapter.common.ProtocolContract
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+import java.util.concurrent.CancellationException
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+fun LocalAssociationScenario.begin(): MobileWalletAdapterClient {
+    return start().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+fun LocalAssociationScenario.end() {
+    close().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+@Suppress("BlockingMethodInNonBlockingContext")
+class MobileWalletAdapter(
+    private val resultCaller: ActivityResultCaller,
+    private val scenarioType: ScenarioTypes = ScenarioTypes.Local,
+    private val timeout: Int = Scenario.DEFAULT_CLIENT_TIMEOUT_MS,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) {
+    private val contract = LocalAssociationIntentContract()
+    private val resultLauncher = resultCaller.registerForActivityResult(contract) { }
```

Comment:
If I remember correctly, `registerForActivityResult()` needs to be called when the activity is being created (or at least before it's actually started/resumed, something like that).

Is there a risk that developers would instantiate `MobileWalletAdapter` later than this, leading to a potential exception?

Reply:
Let me read the docs on that.

Reply:
Welp...
![SmartSelect_20220713-090703_Brave.jpg](https://user-images.githubusercontent.com/2018851/178780394-a1214a15-b4f2-46ec-bc91-c35a9d2570f6.jpg)